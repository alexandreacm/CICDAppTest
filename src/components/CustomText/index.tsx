import React, { PropsWithChildren } from 'react';
import { Text, View, StyleSheet } from 'react-native';
import light from '../../styles/light';
import dark from '../../styles/dark';

type Props = {
  isDarkMode: boolean;
};

const CustomText: React.FC<PropsWithChildren<Props>> = ({ children, isDarkMode }) => {
  return (
    <View>
      <Text
        style={[
          styles.text,
          { color: isDarkMode ? dark.colors.text : light.colors.text },
        ]}
      >
        {children}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  text: {
    fontSize: 20,
    fontWeight: 'bold',
  },
});

export default CustomText;
