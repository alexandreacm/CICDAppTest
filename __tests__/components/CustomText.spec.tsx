import { render, screen } from "@testing-library/react-native";
import CustomText from "../../src/components/CustomText";

describe('<CustomText />', () => {
    it('should render correctly', () => {
        const { getByText } = render(<CustomText isDarkMode={false}>Hello World</CustomText>);
        const textElement = getByText('Hello World');
        expect(textElement).toBeTruthy();
    });

    it('Should render snapshot correctly', () => {
        const tree = render(<CustomText isDarkMode={false}>Hello World</CustomText>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('Should render text with dark mode styles', () => {
        render(<CustomText isDarkMode={true}>Welcome</CustomText>);

        screen.getByText('Welcome');
    })
});