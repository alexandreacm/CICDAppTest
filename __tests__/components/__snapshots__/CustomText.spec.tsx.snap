// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomText /> Should render snapshot correctly 1`] = `
<View>
  <Text
    style={
      [
        {
          "fontSize": 20,
          "fontWeight": "bold",
        },
        {
          "color": "#000000",
        },
      ]
    }
  >
    Hello World
  </Text>
</View>
`;
